### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

The time is finally here to ditch Travis :tada: :tada: 

Here is a link to the moveit2 PR where I've explained many of the details in this PR: https://github.com/ros-planning/moveit2/pull/396

There is this PR that causes all the same formatting changes: https://github.com/ros-planning/moveit/pull/2557  If that gets merged first I'll rebase this one.  Otherwise if this one is merged first that one can just be closed.

I will make PRs for each of the *-devel branches after this is merged.

### CODE REVIEW

**Comment by rhaschke** [comment]
See my comments to https://github.com/ros-planning/moveit2/pull/396.

**Comment by tylerjw** [comment]
@rhaschke I've updated this based on your feedback.

File: .github/workflows/industrial_ci_action.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      - "[kmn]*-devel"
+
+jobs:
+  industrial_ci:
+    strategy:
+      matrix:
+        env:
+          - IMAGE: 'master-ci'
+            CCOV: true
+            AFTER_RUN_TARGET_TEST: './.ci.prepare_codecov'
```
Comment (rhaschke) [inline .github/workflows/industrial_ci_action.yml:23]: This should become a computed variable in the `env` section below:
`AFTER_RUN_TARGET_TEST: ${{ matrix.env.CCOV && './.ci.prepare_codecov' || ''}}`

File: .github/workflows/industrial_ci_action.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+          - IMAGE: 'master-ci-shadow-fixed'
+            CATKIN_LINT: true
+          - IMAGE: 'melodic-ci'
+            BEFORE_TARGET_TEST_EMBED: 'set +u && source moveit_kinematics/test/test_ikfast_plugins.sh && set -u'
+            CLANG_TIDY: true
+    env:
+      DOCKER_IMAGE: moveit/moveit:${{ matrix.env.IMAGE }}
+      UPSTREAM_WORKSPACE: moveit.rosinstall
+      CCOV: ${{ matrix.env.CCOV }}
+      AFTER_SETUP_UPSTREAM_WORKSPACE: 'vcs pull $BASEDIR/upstream_ws/src'
```
Comment (rhaschke) [inline .github/workflows/industrial_ci_action.yml:33]: reorder to group semantically similar things:
```suggestion
      UPSTREAM_WORKSPACE: moveit.rosinstall
      AFTER_SETUP_UPSTREAM_WORKSPACE: 'vcs pull $BASEDIR/upstream_ws/src'
      CCOV: ${{ matrix.env.CCOV }}
```

**Review by rhaschke — COMMENTED** [review]
Sorry, I have some more nits.

**Comment by tylerjw** [comment]
@rhaschke pushed those changes.

**Review by rhaschke — APPROVED** [review]
Great work. I have one more idea: What do you think about moving catkin_lint from industrial_ci into `format.yaml`, i.e. handling it the same way as `clang-format`? This would include it into the local pre-commit check as well. Of course this requires people to install catkin_lint (alongside clang-format). But they should be able to do so, if the can install pre-commit locally.
Otherwise I approve.

File: .github/workflows/industrial_ci_action.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        uses: codecov/codecov-action@v1
+        if: ${{ matrix.env.CCOV }}
+        with:
+          files: ${{ env.BASEDIR }}/coverage.info
+      - name: prepare target_ws for cache
+        if: ${{ always() && ! matrix.env.CCOV }}
+        run: |
+          du -sh ${{ env.BASEDIR }}/target_ws
+          sudo find ${{ env.BASEDIR }}/target_ws -wholename '*/test_results/*' -delete
+          sudo rm -rf ${{ env.BASEDIR }}/target_ws/src
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply .github/workflows/industrial_ci_action.yml:104]: yup, thank you docker

**Comment by tylerjw** [comment]
> Great work. I have one more idea: What do you think about moving catkin_lint from industrial_ci into `format.yaml`, i.e. handling it the same way as `clang-format`? This would include it into the local pre-commit check as well. Of course this requires people to install catkin_lint (alongside clang-format). But they should be able to do so, if the can install pre-commit locally.
> Otherwise I approve.

I like this idea, do you mind if I work on this in a follow-on PR?

File: .github/workflows/industrial_ci_action.yml [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      AFTER_RUN_TARGET_TEST: ${{ matrix.env.CCOV && './.ci.prepare_codecov' || '' }}
+      TARGET_CMAKE_ARGS: >
+        -DCMAKE_BUILD_TYPE=${{ matrix.env.CCOV && 'RelWithDebInfo' || 'Release'}}
+        ${{ matrix.env.CCOV && '-DCMAKE_CXX_FLAGS="--coverage" --no-warn-unused-cli' || '' }}
+      CCACHE_DIR: "${{ github.workspace }}/.ccache"
+      BASEDIR: ${{ github.workspace }}/.work
+      CACHE_PREFIX: "${{ matrix.env.IMAGE }}${{ matrix.env.CCOV && '-ccov' || '' }}"
+      CLANG_TIDY_BASE_REF: "${{ github.base_ref || github.ref }}"
+
+    name: "${{ matrix.env.IMAGE }}${{ matrix.env.CATKIN_LINT && ' + catkin_lint' || ''}}${{ matrix.env.CCOV && ' + ccov' || ''}}${{ contains(matrix.env.IKFAST_TEST, 'ikfast') && ' + ikfast' || ''}}${{ matrix.env.CLANG_TIDY && ' + clang-tidy' || '' }}"
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply .github/workflows/industrial_ci_action.yml:42]: rebased this change in, good catch

**Comment by tylerjw** [comment]
I'll have to think about the catkin_lint thing more and test to see if I can get it to work with pre-commit.  It doesn't quite work like the other linters used by pre-commit in that it can't be pointed at specific files to lint, it has to run over whole directories.

**Comment by rhaschke** [comment]
> I like this idea, do you mind if I work on this in a follow-on PR?

Sure. Let's merge this one here. I will take a look for the Travis issue...

**Comment by tylerjw** [comment]
I created this issue: https://github.com/ros-planning/moveit/issues/2566 to track working on pre-commit.

**Pull request merged**